pkg/cmd/cli/secrets/source.go

```diff
@@ -0,0 +1,198 @@
+package secrets
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	cmdutil "github.com/openshift/origin/pkg/cmd/util"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	CreateSourceSecretRecommendedCommandName = "new-sourcecfg"
+
+	createSourceSecretLong = `
+Create a new source secret
+
+Source secrets are used to authenticate against SCM servers.
+
+When creating applications, you may have a SCM server that requires authentication. In order for the
+nodes to clone source code on your behalf, they have to have the credentials. You can provide this
+information by creating a sourcecfg secret and attaching it to your service account.
+
+If your source authentication method requires only username and password or token, add it by using:
```

Comment:
These should be examples in a const `createSourceSecretExample` instead of part of the long description:
https://github.com/openshift/origin/blob/master/docs/cli_hacking_guide.adoc#commented-example
https://github.com/openshift/origin/blob/master/docs/cli_hacking_guide.adoc#writing-examples